internal/core/src/segcore/SegmentSealedImpl.cpp

```diff
@@ -1481,13 +1481,17 @@ SegmentSealedImpl::mask_with_timestamps(BitsetType& bitset_chunk,
 
 bool
 SegmentSealedImpl::generate_binlog_index(const FieldId field_id) {
-    if (col_index_meta_ == nullptr)
+    std::unique_lock lck(mutex_);
```

Comment:
mutex_ can be locked just before replacing filedIndexing(line 1542) for better performance (parallel loading).